pygmt/base_plotting.py

```diff
@@ -658,7 +659,9 @@ def legend(self, spec=None, **kwargs):
                 specfile = spec
             else:
                 raise GMTInvalidInput("Unrecognized data type: {}".format(type(spec)))
-            arg_str = " ".join([specfile, build_arg_string(kwargs)])
+            arg_str = " ".join(
+                [specfile, "-D{}".format(position), build_arg_string(kwargs)]
+            )
```

Comment:
Oo, the thing which this is that if someone were to *not* use the alias `position`, e.g. they use `fig.legend(D="JTR+jTR")` instead, then they'll get an error. I know we're trying to discourage using single-character arguments but I'm still a bit hesitant on this...

Reply:
Whoops. Good point. I don't immediately see a work-around here.

Reply:
Okay, now we check for the presence of the `'D'` key in pre-processed`kwargs`. This key is present if the user used `D` **OR** `position`, since the kwargs have been processed already. If the key isn't present, we use the default `position = 'JTR+jTR'`.